examples/c/ex_tiered.c

```diff
@@ -0,0 +1,188 @@
+ * specific cloud provider). The dir_store storage source stores objects in a directory named by the
+ * bucket name, relative to the WiredTiger home directory.
+ */
+#define BUCKET_NAME "bucket"
+#define BUILD_DIR "../../"
+#define STORAGE_SOURCE "dir_store"
+
+int
+main(int argc, char *argv[])
+{
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+    const char *home;
+    char config[1024], bucketdir[1024];
+
+    home = example_setup(argc, argv);
+
+    /*
+     * Set up configuration for tiered storage. When tiered storage is configured, all tables
+     * created will be tiered by default (that is, parts will be stored in object storage), though
+     * tiered storage can be turned off on an individual basis.
+     *
+     * When configuring tiered storage with a cloud provider, one usually needs an access key. A
+     * reference to this, an "authorization token", is usually provided in the configuration string.
+     * This simple example uses "dir_store", a local directory based stand-in for a cloud provider,
+     * and no explicit access keys are needed.
+     *
+     * For learning purposes, one can single step this program and watch files created in the
+     * "bucket" subdirectory during the tiered checkpoint.
+     */
+    snprintf(config, sizeof(config),
+      "create,"
+      "tiered_storage=(bucket=%s,bucket_prefix=pfx-,local_retention=5,name=%s),"
+      "extensions=(%s/ext/storage_sources/%s/libwiredtiger_%s.so)",
+      BUCKET_NAME, STORAGE_SOURCE, BUILD_DIR, STORAGE_SOURCE, STORAGE_SOURCE);
+
+    /* Create the home directory, and the bucket directory underneath it. */
+    if (access(home, R_OK | W_OK | X_OK) != 0)
+        error_check(mkdir(home, 0777));
+    snprintf(bucketdir, sizeof(bucketdir), "%s/%s", home, BUCKET_NAME);
+    if (access(bucketdir, R_OK | W_OK | X_OK) != 0)
+        error_check(mkdir(bucketdir, 0777));
+
+    /* Configure the connection to use tiered storage. */
+    error_check(wiredtiger_open(home, NULL, config, &conn));
+
+    /* Open a session for the current thread's work. */
+    error_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Create a table that lives locally. Tiered storage is disabled for this file. */
+    error_check(session->create(
+      session, "table:housecat", "key_format=i,value_format=S,tiered_storage=(name=none)"));
+
+    /* Create a table using tiered storage. */
+    error_check(session->create(session, "table:wildcat", "key_format=i,value_format=S"));
+
+    /* Add entries to both tables. */
+    add_data(session, "table:housecat", 0, 10);
+    add_data(session, "table:wildcat", 0, 10);
+
+    /*
+     * Do a regular checkpoint. Checkpoints are usually done in their own thread with their own
+     * session. Data is synchronized to local storage.
+     */
+    error_check(session->checkpoint(session, NULL));
+
+    /* Add more entries to both tables. */
+    add_data(session, "table:housecat", 10, 10);
```

Comment:
Similarly, the first key should be `NUM_ENTRIES` and the next one `2 * NUM_ENTRIES` so that if someone wants to add a million keys each time, it all just works.